cdc/puller/frontier/frontier.go

```diff
@@ -63,33 +76,50 @@ func (s *spanFrontier) Frontier() uint64 {
 }
 
 // Forward advances the timestamp for a span.
-func (s *spanFrontier) Forward(span regionspan.ComparableSpan, ts uint64) {
-	s.insert(span, ts)
+func (s *spanFrontier) Forward(regionID uint64, span regionspan.ComparableSpan, ts uint64) {
+	// it's the fast part to detect if the region is split or merged,
+	// if not we can update the minTsHeap with use new ts directly
+	if n, ok := s.cachedRegions[regionID]; ok && n.regionID > 0 && n.end != nil {
+		if regionID == n.regionID && bytes.Equal(n.Key(), span.Start) && bytes.Equal(n.End(), span.End) {
+			s.minTsHeap.UpdateKey(n.Value(), ts)
+			s.metricResolvedRegionCachedCounter.Inc()
```

Comment:
If we make the assumption that `region merge or split is relatively rare`, L82 ~ L85 should be hit frequently, which makes L85 always be called.

Can we just record `cache miss` case to reduce the overhead caused by updating the metrics?

Reply:
done